src/ami-set.js

```diff
@@ -51,6 +55,49 @@ AmiSet.listAmiSets = async function () {
 };
 
 /**
+ * Checks if the AMIs from the amiSet are valid or not.
+ */
+
+async function checkAmi(ctx, region, ami, vtype) {
```

Comment:
So, given that you're passing in the scope anyway through `ctx`, let's make this an extra `AmiSet` method.  `AmiSet.__checkAmi(region, ami, vtype)` would be great.